pkgs/development/libraries/ncurses/default.nix

```
@@ -48,20 +48,18 @@ stdenv.mkDerivation (finalAttrs: {
         "/usr/share/terminfo" # upstream default, probably all FHS-based distros
         "/run/current-system/sw/share/terminfo" # NixOS
       ]}"
-  ];
+  ] ++ lib.optional (stdenv.buildPlatform != stdenv.hostPlatform) "--with-build-cc=${stdenv.cc.targetPrefix}cc";
```

I tried building `pkgsCross.aarch64-multiplatform.ncurses` with this PR and it didn't work.

But with the following change it did:

```suggestion
  ] ++ lib.optionals (stdenv.buildPlatform != stdenv.hostPlatform) [
    "--with-build-cc=${buildPackages.stdenv.cc}/bin/${buildPackages.stdenv.cc.targetPrefix}cc"
  ];
```

Note that it's okay to use `buildPackages` here because this is isn't `depsFooBar`, `buildInputs`, or `nativeBuildInputs`.